src/abstractdatatable/reshape.jl

```diff
@@ -193,9 +193,23 @@ function unstack(dt::AbstractDataTable, rowkey::Int, colkey::Int, value::Int)
     # `rowkey` integer indicating which column to place along rows
     # `colkey` integer indicating which column to place along column headers
     # `value` integer indicating which column has values
-    refkeycol = NullableCategoricalArray(dt[rowkey])
+    if typeof(dt[rowkey]) <: CategoricalVector
```

Comment:
It works for Vectors but not CategoricalVectors

```julia
julia> a = CategoricalVector(["a", "b", "c"])
3-element CategoricalArrays.CategoricalArray{String,1,UInt32,String,Union{}}:
 "a"
 "b"
 "c"

julia> b = CategoricalVector(["a", "b", null])
3-element CategoricalArrays.CategoricalArray{?String,1,UInt32,String,Nulls.Null}:
 "a"
 "b"
 null

julia> c = ["a", "b", "c"]
3-element Array{String,1}:
 "a"
 "b"
 "c"

julia> d = ["a", "b", null]
3-element Array{?String,1}:
 "a"
 "b"
 null

julia> CategoricalVector{?eltype(a)}(a)
ERROR: MethodError: Cannot `convert` an object of type String to an object of type CategoricalArrays.CategoricalValue{String,UInt32}
This may have arisen from a call to the constructor CategoricalArrays.CategoricalValue{String,UInt32}(...),
since type constructors fall back to convert methods.
Stacktrace:
 [1] setindex!(::Array{CategoricalArrays.CategoricalValue{String,UInt32},1}, ::String, ::Int64) at ./array.jl:615
 [2] copy! at ./abstractarray.jl:654 [inlined]
 [3] copy! at ./abstractarray.jl:646 [inlined]
 [4] convert at ./array.jl:404 [inlined]
 [5] convert(::Type{CategoricalArrays.CategoricalPool{CategoricalArrays.CategoricalValue{String,UInt32},UInt32,V} where V}, ::CategoricalArrays.CategoricalPool{String,UInt32,CategoricalArrays.CategoricalValue{String,UInt32}}) at /Users/Cameron/.julia/v0.7/CategoricalArrays/src/pool.jl:70
 [6] convert(::Type{CategoricalArrays.CategoricalArray{?CategoricalArrays.CategoricalValue{String,UInt32},1,UInt32,V,U} where U where V}, ::CategoricalArrays.CategoricalArray{String,1,UInt32,String,Union{}}) at /Users/Cameron/.julia/v0.7/CategoricalArrays/src/array.jl:284
 [7] #call#25(::Bool, ::Type{CategoricalArrays.CategoricalArray{?CategoricalArrays.CategoricalValue{String,UInt32},1,UInt32,V,U} where U where V}, ::CategoricalArrays.CategoricalArray{String,1,UInt32,String,Union{}}) at /Users/Cameron/.julia/v0.7/CategoricalArrays/src/array.jl:173
 [8] (::Core.#kw#Type)(::Array{Any,1}, ::Type{CategoricalArrays.CategoricalArray{?CategoricalArrays.CategoricalValue{String,UInt32},1,UInt32,V,U} where U where V}, ::CategoricalArrays.CategoricalArray{String,1,UInt32,String,Union{}}) at ./<missing>:0
 [9] CategoricalArrays.CategoricalArray{?CategoricalArrays.CategoricalValue{String,UInt32},1,V,U,U} where U where U where V(::CategoricalArrays.CategoricalArray{String,1,UInt32,String,Union{}}) at /Users/Cameron/.julia/v0.7/CategoricalArrays/src/array.jl:217

julia> CategoricalVector{?eltype(b)}(b)
ERROR: MethodError: Cannot `convert` an object of type String to an object of type CategoricalArrays.CategoricalValue{String,UInt32}
This may have arisen from a call to the constructor CategoricalArrays.CategoricalValue{String,UInt32}(...),
since type constructors fall back to convert methods.
Stacktrace:
 [1] setindex!(::Array{CategoricalArrays.CategoricalValue{String,UInt32},1}, ::String, ::Int64) at ./array.jl:615
 [2] copy! at ./abstractarray.jl:654 [inlined]
 [3] copy! at ./abstractarray.jl:646 [inlined]
 [4] convert at ./array.jl:404 [inlined]
 [5] convert(::Type{CategoricalArrays.CategoricalPool{CategoricalArrays.CategoricalValue{String,UInt32},UInt32,V} where V}, ::CategoricalArrays.CategoricalPool{String,UInt32,CategoricalArrays.CategoricalValue{String,UInt32}}) at /Users/Cameron/.julia/v0.7/CategoricalArrays/src/pool.jl:70
 [6] convert(::Type{CategoricalArrays.CategoricalArray{?CategoricalArrays.CategoricalValue{String,UInt32},1,UInt32,V,U} where U where V}, ::CategoricalArrays.CategoricalArray{?String,1,UInt32,String,Nulls.Null}) at /Users/Cameron/.julia/v0.7/CategoricalArrays/src/array.jl:284
 [7] #call#25(::Bool, ::Type{CategoricalArrays.CategoricalArray{?CategoricalArrays.CategoricalValue{String,UInt32},1,UInt32,V,U} where U where V}, ::CategoricalArrays.CategoricalArray{?String,1,UInt32,String,Nulls.Null}) at /Users/Cameron/.julia/v0.7/CategoricalArrays/src/array.jl:173
 [8] (::Core.#kw#Type)(::Array{Any,1}, ::Type{CategoricalArrays.CategoricalArray{?CategoricalArrays.CategoricalValue{String,UInt32},1,UInt32,V,U} where U where V}, ::CategoricalArrays.CategoricalArray{?String,1,UInt32,String,Nulls.Null}) at ./<missing>:0
 [9] CategoricalArrays.CategoricalArray{?CategoricalArrays.CategoricalValue{String,UInt32},1,V,U,U} where U where U where V(::CategoricalArrays.CategoricalArray{?String,1,UInt32,String,Nulls.Null}) at /Users/Cameron/.julia/v0.7/CategoricalArrays/src/array.jl:217

julia> CategoricalVector{?eltype(c)}(c)
3-element CategoricalArrays.CategoricalArray{?String,1,UInt32,String,Nulls.Null}:
 "a"
 "b"
 "c"

julia> CategoricalVector{?eltype(d)}(d)
3-element CategoricalArrays.CategoricalArray{?String,1,UInt32,String,Nulls.Null}:
 "a"
 "b"
 null
```

We could address it in CategoricalArrays, or at the very least I could move this to a helper function to cut down on repeating these lines 3x.